emcc.py

```diff
@@ -1911,7 +1924,7 @@ def get_clang_command(input_files):
         # output the dependency rule. Warning: clang and gcc behave differently
         # with -MF! (clang seems to not recognize it)
         logger.debug(('just preprocessor ' if has_dash_E else 'just dependencies: ') + ' '.join(cmd))
-        return run_process(cmd, check=False).returncode
+        return run_process(cmd, check=False, input=pipe_stdin).returncode
```

Comment:
I don't think you need to do this.   Python's subprocess will be default pass the stdin file descriptor to the subprocess.  So I think you can simply remove `pipe_stdin` and let clang's handling of `-c -` kick in.

In short, all you need to do has have `-` added to input_files and you should be done.

Reply:
Its also more efficient not to read the whole of stdin here, but let clang read it on demand.

Reply:
Yes, Indeed. I was wondering how to pipe contents of stdin to subprocess. it seems to happen by default.